web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -74,58 +74,64 @@ public JsPartitionedTable(WorkerConnection connection, JsWidget widget) {
 
     @JsIgnore
     public Promise<JsPartitionedTable> refetch() {
-        if (keys != null) {
-            keys.close();
-        }
-        if (subscription != null) {
-            subscription.close();
-        }
+        closeSubscriptions();
         return widget.refetch().then(w -> {
             descriptor = PartitionedTableDescriptor.deserializeBinary(w.getDataAsU8());
 
             return w.getExportedObjects()[0].fetch();
-        }).then(result -> {
-            keys = (JsTable) result;
+        }).then(result -> connection.newState((c, state, metadata) -> {
+            final JsTable rawKeyTable = (JsTable) result;
+            DropColumnsRequest drop = new DropColumnsRequest();
+            drop.setColumnNamesList(new String[] {descriptor.getConstituentColumnName()});
+            drop.setSourceId(rawKeyTable.state().getHandle().makeTableReference());
+            drop.setResultId(state.getHandle().makeTicket());
+            connection.tableServiceClient().dropColumns(drop, metadata, c::apply);
+            rawKeyTable.close();
```

Comment:
This might race and close the rawKeyTable's ticket before the drop columns above finishes.

The array-of-one (or atomic reference) thing is a valid fix, you could also expand `c::apply` into a lambda and directly call `rawKeyTable.close()` in there.

Reply:
I guess technically expanding `c::apply` to be a lambda could be incorrect if the dropColumns operation fails; the table wouldn't get closed then. I've put the `.close` in a `finally` block.

Reply:
You could, but c::apply handles both success and failure - its one of my favorite design decisions of this grpc-web library, that the args to the callback are (failure, success), so that it is that much harder to forget to handle failure cases.

Reply:
(if c::apply didnt handle failure, neither .then/catch/finally would get invoked anyway, since nothing else resolves/rejects the promise)

Reply:
Only if you want to?

```
            connection.tableServiceClient().dropColumns(drop, metadata, (fail, success) -> {
                rawKeyTable.close();
                c.apply(fail, success);
            });
```
That would let you skip the array/atomicref, and just close the table after the dropcols is complete, pass or fail, and regardless of pass/fail, propagate that status onwards.